test/generate_ssl_tests.pl

```diff
@@ -43,6 +43,7 @@ sub print_templates {
     # Add the implicit base configuration.
     foreach my $test (@ssltests::tests) {
         $test->{"server"} = { (%ssltests::base_server, %{$test->{"server"}}) };
+	$test->{"server2"} = { (%ssltests::base_server, %{$test->{"server2"}}) };
```

Comment:
Good point, we already have a SNI indicator!

So I think that you should be able to test `defined $test->{"server2"}` here and only merge in the basic conf if "server2" exists. 

Then in the template, do a similar test for %server2 and omit the output if undef. Does that work?